### PR TITLE
Add gandi DNS service

### PR DESCRIPTION
**Pull request by raphink**
This adds the gandi DNS service as implemented in https://github.com/rancher/external-dns/pull/27

### CODE REVIEW

File: templates/gandi/0/rancher-compose.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.catalog:
+  name: "Gandi DNS"
+  version: "v0.2.1-rancher1"
```
Comment (alena1108) [inline templates/gandi/0/rancher-compose.yml]: Version should be v0.3.0

File: templates/gandi/config.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name: Gandi DNS
+description: |
+  Rancher External DNS service powered by Gandi
+version: v0.1.0-rancher1
```
Comment (alena1108) [inline templates/gandi/config.yml]: v0.3.0 :)

File: templates/gandi/0/rancher-compose.yml (hunk 1 above)
Comment (raphink) [inline templates/gandi/0/rancher-compose.yml]: ah yes, my bad.

Comment (raphink) [inline templates/gandi/0/rancher-compose.yml]: I thought this was the catalog service version, not the container version (which is 0.3.0)

**Comment by raphink** [comment]
@alena1108 there, I tagged everything 0.3.0

File: templates/gandi/catalogIcon-gandi.svg [hunk 3]
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
Comment (raphink) [inline templates/gandi/catalogIcon-gandi.svg:2]: Haha, inkscape didn't remove that :smile:

File: templates/gandi/0/rancher-compose.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  questions:
+    - variable: "GANDI_APIKEY"
+      label: "Gandi API key"
+      description: "API key Gandi account"
+      type: "string"
+      required: true
+    - variable: "GANDI_TESTING"
+      label: "Testing environment"
+      description: "Whether use the test platform (OT&E)"
+      type: "string"
```
Comment (raphink) [inline templates/gandi/0/rancher-compose.yml:16]: @alena1108 I was actually wondering if there was a type `bool` for this, that would set it to true if checked. Is that supported already?

Comment (alena1108) [inline templates/gandi/0/rancher-compose.yml:16]: @prachidamle I believe we do support boolean types in our templates?

Comment (prachidamle) [inline templates/gandi/0/rancher-compose.yml:16]: @raphink @alena1108 yup, boolean is supported. 
Full list of types supported here: http://docs.rancher.com/rancher/catalog/

Comment (raphink) [inline templates/gandi/0/rancher-compose.yml:16]: With a boolean value, will the passed value be interpreted in go as a boolean or a string "true"/"false"?

**Comment by deniseschannon** [comment]
@raphink We've moved everything contributed and maintained by the community to 
https://github.com/rancher/community-catalog

Can you please open up a new PR in that repo?

**Comment by raphink** [comment]
@deniseschannon thanks. I'm actually not using this anymore, as Gandi cannot garantee DNS changes under 20 minutes (when all goes well), so it's no good fit for external DNS.

**Pull request closed without merging**